src/Campaigns/resources/admin/components/CampaignDetailsPage/index.tsx

```diff
@@ -1,214 +1,144 @@
+        record: Campaign,
+        hasResolved: boolean,
+        save: () => any,
+        edit: (data: Campaign) => void,
+    } = useEntityRecord('givewp', 'campaign', campaignId);
 
-    const methods = useForm<CampaignDetailsInputs>({
-        defaultValues: {
-            title: campaign.properties.title ?? '',
-        },
+    const methods = useForm<Campaign>({
+        mode: 'onChange',
+        ...resolver,
     });
-    const {formState, handleSubmit, watch} = methods;
-    const formWatch = watch();
-
-    useEffect(() => {
-        console.log('formWatch: ', formWatch);
-        console.log('formState.dirtyFields: ', formState.dirtyFields);
-        console.log('formState.isDirty: ', formState.isDirty);
-    }, [formWatch]);
 
-    const onSubmit: SubmitHandler<CampaignDetailsInputs> = async (campaignDetailsInputs, event) => {
-        event.preventDefault();
+    const {formState, handleSubmit, reset, setValue} = methods;
 
-        try {
-            if (isPublishMode) {
-                console.log('publishing...');
-                const endpoint = `/${campaign.properties.id}/publish`;
-                const response = await API.fetchWithArgs(endpoint, {}, 'PUT');
-                console.log('Campaign published.', response);
-                location.reload();
-            } else if (formState.isDirty) {
-                console.log('updating...');
-                const endpoint = `/${campaign.properties.id}`;
-                const response = await API.fetchWithArgs(endpoint, campaignDetailsInputs, 'PUT');
-                console.log('Campaign updated.', response);
-                location.reload();
-            }
-        } catch (error) {
-            console.error('Error updating campaign.', error);
+    // Set default values when campaign is loaded
+    useEffect(() => {
+        if (hasResolved) {
+            reset({...campaign});
+        }
+    }, [hasResolved]);
+
+    const onSubmit: SubmitHandler<Campaign> = async (data, e) => {
+        e.preventDefault();
+
+        if (formState.isDirty) {
+            edit(data);
+
+            save()
+                .then((response: Campaign) => {
+                    reset(response);
+                })
+                .catch((response: any) => {
+                    //todo: add error handling
+                    console.log(response);
+                });
         }
     };
 
+    if (!hasResolved) {
+        return (
+            <div className={styles.loadingContainer}>
+                <div className={styles.loadingContainerContent}>
+                    <Spinner />
+                    <div className={styles.loadingContainerContentText}>
+                        {__('Loading campaign...', 'give')}
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
     return (
         <FormProvider {...methods}>
             <form onSubmit={handleSubmit(onSubmit)}>
                 <article className={styles.page}>
```

Comment:
Currently, the `ClassicEditor` component is throwing some errors in the browser console because it was designed to be used inside the Gutenberg editor at the first moment. 

So it's looking for an element in the DOM with the `.interface-interface-skeleton__content` class necessary to make [this workaround](https://github.com/impress-org/form-builder-library/blob/master/src/ClassicEditor/index.tsx#L109-L127) work. 

To prevent these errors we can change this line to something like this:

```
<article className={`interface-interface-skeleton__content ${styles.page}`}>
```

This way, we can use the `ClassicEditor` component anywhere on the campaign details page without throwing errors in the console.

Reply:
@glaubersilva it works, awesome. Thanks, man!